backup-stores/azure/src/test/java/io/camunda/zeebe/backup/azure/AzureBackupStoreIT.java

```diff
@@ -77,8 +97,48 @@ void backupShouldExistAfterStoreIsClosed(final Backup backup) {
     assertThat(status.lastModified()).isEqualTo(firstStatus.lastModified());
   }
 
-  @Test
-  void cannotDeleteUploadingBlock() {
-    // TODO: when delete feature is done
+  @ParameterizedTest
+  @ArgumentsSource(TestBackupProvider.class)
+  public void cannotDeleteUploadingBlock(final Backup backup) {
+
+    // given when
+    uploadInProgressManifest(backup);
+
+    // then
+    Assertions.assertThat(getStore().delete(backup.id()))
+        .failsWithin(Duration.ofSeconds(10))
+        .withThrowableOfType(Throwable.class)
+        .withRootCauseInstanceOf(UnexpectedManifestState.class);
+  }
+
+  void uploadInProgressManifest(final Backup backup) {
+    final var manifest = Manifest.createInProgress(backup);
+    final byte[] serializedManifest;
+
+    try {
+      serializedManifest = MAPPER.writeValueAsBytes(manifest);
+    } catch (final JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
+
+    final BlobClient blobClient = buildBlobClient(manifest);
+    blobClient.upload(BinaryData.fromBytes(serializedManifest));
+  }
+
+  BlobClient buildBlobClient(final Manifest manifest) {
+    final BlobServiceClient blobServiceClient =
+        new BlobServiceClientBuilder()
+            .connectionString(azureBackupConfig.connectionString())
+            .buildClient();
+    final BlobContainerClient blobContainerClient =
+        blobServiceClient.getBlobContainerClient(azureBackupConfig.containerName());
+    blobContainerClient.createIfNotExists();
+    return blobContainerClient.getBlobClient(manifestPath(manifest));
+  }
+
+  String manifestPath(final Manifest manifest) {
+    final String manifestPathFormat = "%s/%s/%s/manifest.json";
+    return manifestPathFormat.formatted(
+        manifest.id().partitionId(), manifest.id().checkpointId(), manifest.id().nodeId());
```

Comment:
:upside_down_face: You could also decide to reuse the same method from `ManifestManager`